scripts/xblock/xblock_counts.py

```diff
@@ -0,0 +1,367 @@
+BLOCKS_URL_KEY = 'blocks_url'
+BLOCK_ROOT_KEY = 'root'
+BLOCKS_KEY = 'blocks'
+BLOCK_COUNTS_KEY = 'block_counts'
+COURSE_NAME_KEY = 'name'
+COURSE_ID_KEY = 'course_id'
+COURSE_START_KEY = 'start'
+COURSE_END_KEY = 'end'
+
+
+def monthdelta(date, delta):
+    """
+    Method to get a delta of Months from a provided datetime
+
+    From this StackOverflow response:
+    http://stackoverflow.com/questions/3424899/whats-the-simplest-way-to-subtract-a-month-from-a-date-in-python
+
+    Arguments:
+        date datetime: Date to be modified
+        delta int: delta value
+
+    Returns:
+        datetime: The datetime with the month delta applied
+    """
+    m, y = (date.month + delta) % 12, date.year + (date.month + delta - 1) // 12
+    if not m:
+        m = 12
+    d = min(date.day, [31,
+                       29 if y % 4 == 0 and not y % 400 == 0
+                       else 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31][m - 1])
+    return date.replace(day=d, month=m, year=y)
+
+
+def _get_course_data_summary(auth_token, months_restriction, xblock_type_set, api_root, course_count=None):
+    """
+    Gets the course summary data from the Course Blocks API and returns a list of data objects
+    summarizing each courses xBlock usage
+
+    Arguments
+        auth_token (str): Authentication token for the API
+        months_restriction (int): Restriction on the number of months to go back
+        xblock_type_set (set): A set of Strings containing the xBlocks types to be counted
+
+    Returns:
+        list: a list of data objects summarizing each courses xBlock usage
+    """
+    # Get the Course list
+    response = requests.get(api_root + '/api/courses/v1/courses/')
+    json_result = response.json()
+
+    num_pages = 1
+    if PAGINATION_KEY in json_result and NUM_PAGES_KEY in json_result[PAGINATION_KEY]:
+        num_pages = json_result[PAGINATION_KEY][NUM_PAGES_KEY]
+        num_courses = json_result[PAGINATION_KEY]['count']
+
+    course_summary_data = []
+    block_type_url = _get_block_count_url_string(xblock_type_set)
+
+    if course_count is None:
+        course_count = num_courses
+
+    course_count_limit = False
+    total_courses = 0
+    # Look through all pages and courses
+    while num_pages > 0 and not course_count_limit:
+        if RESULTS_KEY in json_result:
+            results_list = json_result[RESULTS_KEY]
+            for course in results_list:
+                course_data = _get_course_data(auth_token, course, block_type_url,
+                                               months_restriction=months_restriction)
+                if course_data is not None:
+                    course_summary_data.append(course_data)
+
+                if total_courses == course_count:
+                    course_count_limit = True
+                    break
+                total_courses += 1
+        num_pages -= 1
+        sys.stdout.write('.')
+        sys.stdout.flush()
+    print 'Processed %d courses' % total_courses
```

Comment:
Nit (that might be wrong): Just from generally observing our codebase I think `'Processed {} courses'.format(total_courses)` would be the favored way to do this string formatting.